### PR TITLE
Gsl testcases

### PR DESCRIPTION
**Pull request by vijay19482**
1. Replace _xxx_to_txt functions  with ffi.cast in PRINT completed. All the testcases are running after this change. 

2. Clean up of MK_COL,LOAD,PRINT,METADATA,DATALOAD with respect to below pattern is done

1) Do not define functions as 
function foo () 
end

Instead
return function ()
end

3. PRINT implementation , will return string , if filename not provided. Also, it will be able to handle Column as input.

### CODE REVIEW

File: OPERATORS/DATA_LOAD/test/test_load.lua [hunk 1]
```diff
@@ -23,7 +23,7 @@ local srcs = {  "../src/get_cell.c",
 local tgt = "../obj/libload_csv.so"
 local status = assert(compile_so(incs, srcs, tgt), "compile of .so failed")
 
-require 'load_csv_dataload'
+local load_csv = require 'load_csv_dataload'
```
Comment (itsindra) [inline OPERATORS/DATA_LOAD/test/test_load.lua]: what about utils? Is that the one Ramesh wrote?

File: OPERATORS/DATA_LOAD/test/testcases/handle_category.lua [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-require 'C_to_txt'
+local convert_c_to_txt = require 'C_to_txt'
 
 local number_of_testcases_passed = 0
 local number_of_testcases_failed = 0
 
 local failed_testcases = {}
 
+local fns = {}
+
```
Comment (itsindra) [inline OPERATORS/DATA_LOAD/test/testcases/handle_category.lua]: Line 12, or have you assumed that the function is global?

Comment (itsindra) [inline OPERATORS/DATA_LOAD/test/testcases/handle_category.lua]: Line 57 too

File: OPERATORS/DATA_LOAD/test/testcases/test_load_csv.lua [hunk 3]
```diff
@@ -43,21 +28,21 @@ for i, v in ipairs(T) do
   local M = dofile(test_metadata_dir..v.meta)
   local D = v.data
   -- if category6 then set environment in handle_input_category6 function
-  if v.category == "category6" then
-    --print(v.input_regex)
-    handle_input_category6(v.input_regex)
-  end
+  local key = "handle_input_"..v.category
+    if fns[key] then
+      F = fns[key](v.input_regex)
```
Comment (itsindra) [inline OPERATORS/DATA_LOAD/test/testcases/test_load_csv.lua]: and what if you dont have a function for that category. Is that an error?

File: OPERATORS/LOAD_CSV/lua/validate_meta.lua [hunk 4]
```diff
@@ -4,7 +4,7 @@ local pl = require 'pl'
 local dbg = require 'debugger'
 local plpath = require 'pl.path'
   
-function validate_meta(
+return function (
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/lua/validate_meta.lua]: consider making error_code local too

File: OPERATORS/LOAD_CSV/test/testcases/test_load_csv.lua [hunk 5]
```diff
@@ -11,21 +11,6 @@ _G["Q_DICTIONARIES"] = {}
 dir.makepath(_G["Q_DATA_DIR"])
```
Comment (itsindra) [inline OPERATORS/LOAD_CSV/test/testcases/test_load_csv.lua]: where is variable dir created?

File: OPERATORS/DATA_LOAD/test/testcases/test_load_csv.lua (hunk 3 above)
Comment (vijay19482) [inline OPERATORS/DATA_LOAD/test/testcases/test_load_csv.lua]: if function for a category, is not defined, then that testcase is marked failed with the reason  --- handle function for this category not defined.

**Pull request merged**